backend/inspirehep/matcher/utils.py

```diff
@@ -59,21 +53,42 @@ def map_refextract_to_schema(extracted_references, source=None):
     return result
 
 
-@contextmanager
-def retrieve_uri(uri, outdir=None):
-    """Retrieves the given uri and stores it in a temporary file."""
-    with tempfile.NamedTemporaryFile(
-        prefix="inspire", dir=outdir
-    ) as local_file, fsopen(uri, mode="rb") as remote_file:
-        copy_file(remote_file, local_file)
+def create_journal_dict():
+    """
+    Returns a dictionary that is populated with refextracts's journal KB from the database.
+
+    Uses two raw DB queries that use syntax specific to PostgreSQL to generate
+    a dictionary in the format that refextract expects as following::
+
+        { SOURCE: DESTINATION }
+
+    which represents that ``SOURCE`` is translated to ``DESTINATION`` when found.
+    """
+    only_journals = type_coerce(RecordMetadata.json, JSONB)["_collections"].contains(
+        ["Journals"]
+    )
+    entity_short_title = RecordMetadata.json["short_title"]
+    entity_journal_title = RecordMetadata.json["journal_title"]["title"]
+    entity_title_variants = RecordMetadata.json["title_variants"]
+
+    titles_query = RecordMetadata.query.with_entities(
+        entity_short_title,
+        entity_journal_title,
+    ).filter(only_journals)
+
+    title_variants_query = RecordMetadata.query.with_entities(
+        entity_short_title,
+        entity_title_variants,
+    ).filter(only_journals)
+
+    title_dict = {}
 
-        local_file.flush()
-        yield local_file.name
+    for (short_title, journal_title) in titles_query.all():
+        title_dict[short_title] = short_title
+        title_dict[journal_title] = short_title
 
+    for (short_title, title_variants) in title_variants_query.all():
+        sub_dict = dict(map(lambda e: (e, short_title), title_variants))
```

Comment:
comprehensions are more readable and idiomatic in Python
```suggestion
        sub_dict = {title_variant: short_title for title_variant in title_variants}
```